src/types/bookmarks.ts

```diff
@@ -0,0 +1,23 @@
+export enum BookmarkTypes {
+  folder = 'folder',
+  shabad = 'file',
+  bani = 'book-open'
+}
+export enum BookmarksNavigatorRoutes {
+  bookmarks = 'BookmarksList'
+}
+export type BookmarksNavigatorParams = {
+  [BookmarksNavigatorRoutes.bookmarks]: {folder?: string, folderData: FolderData[]},
+}
+
+export type FolderData = {
+  folderName: string,
+  type: BookmarkTypes,
+  items: FolderItem[],
```

Comment:
Try to define a recursive type, such that folders can be nested, since I don't believe they can

Reply:
we are doing the nested folder implementation

Reply:
I think what he's saying is that thus far I've advocated for a non nesting folder set up

Can discuss in issue or here in this comment thread

My proposal is similar to YouTube, Instagram, and other popular apps which allows saving/bookmarking: Flat list of folders, each folder has a flat list of items (non-folders)

I believe the browser proposal allowing nested folders is unnecessary and more difficult to manage programmatically and for so called "power users"

Reply:
sorry, my original comment was missing the question mark 😅

Reply:
@bhajneet I think regardless of whether we put a soft limit on the nested folders or not, it shouldn't be extra work to support a nested structure, and this will allow for the use-case that you mentioned previously of having our own predefined "nested" folders for things like Nitnem should we choose not to nest.

Aka, a nested design structure is actually just a good implementation path for even if there was 1 level of nesting.